crates/sui-types/src/transaction.rs

```diff
@@ -1632,8 +1616,6 @@ pub struct TransactionDataV1 {
 
 impl TransactionData {
     fn new_system_transaction(kind: TransactionKind) -> Self {
-        // assert transaction kind if a system transaction
```

Comment:
This feels a bit strange.
I understand the reason we are removing it, since now we could be constructing a system transaction using a normal PTB kind.
But then the existence of this function itself is now questionable, because it allows you to create a system transaction that may not be a system transaction (e.g. a normal PTB with non-zero sender).

Reply:
Well, now any PTB at all with a `0x0` sender is a system transaction. 

I suppose we could add more checks, such as restricting the entry points to only call the framework? But I'm not sure there's any risk of accidentally making a bad system tx?